astroquery/cds/core.py

```diff
@@ -0,0 +1,201 @@
+    # method also has a 'radius' keyword for specifying the radius around
+    # the coordinates in which to search. If the region is a box, then
+    # the keywords 'width' and 'height' should be used instead. The coordinates
+    # may be accepted as an `astropy.coordinates` object or as a string, which
+    # may be further parsed.
+
+    # similarly we write a query_region_async method that makes the
+    # actual HTTP request and returns the HTTP response
+    def query_region(self, constraints, output_format=OutputFormat(), get_query_payload=False):
+        """
+        Query the cds mocserver which gives us all the datasets matching the constraints.
+        The output format of the query response is expressed by the param output_format
+
+        :param constraints: The constraints (spatial and algebraic) needed by the mocserver for
+        getting us all the matching datasets
+        :param output_format: The output format we want the matching datasets (full record, just IDs, mocpy object)
+        :param get_query_payload: specify if we only want the cds service to return the payload
+        :return: matching datasets in a format output_format dependant. (e.g. a list of dataset IDs if we want
+        just the IDs, a massive union moc resulting from the union of all the matching dataset mocs or a
+        dictionary of datasets indexed by their IDs if the user wants the records of the matching datasets).
+        """
+        response = self.query_region_async(constraints, output_format, get_query_payload)
+
+        if get_query_payload:
+            return response
+
+        result = CdsClass._parse_result_region(response, output_format)
+
+        return result
+
+    @staticmethod
+    def _remove_duplicate(value_l):
+        if isinstance(value_l, list):
+            value_l = list(set(value_l))
+            if len(value_l) == 1:
+                return value_l[0]
+
+        return value_l
+
+    def query_region_async(self, constraints, output_format, get_query_payload, cache=True):
+        """
+        Query the cds mocserver which gives us all the datasets matching the constraints.
+        The output format of the query response is expressed by the param output_format
+
+        :param constraints: The constraints (spatial and algebraic) needed by the mocserver for
+        getting us all the matching datasets
+        :param output_format: The output format we want the matching datasets (full record, just IDs, mocpy object)
+        :param get_query_payload: specify if we only want the cds service to return the payload
+        :param cache: boolean
+        :return: The HTTP response returned from the service. All async methods should return the raw HTTP response.
+        """
+        request_payload = dict()
+        if not isinstance(constraints, Constraints):
+            print("Invalid constraints. Must be of MOCServerConstraints type")
+            raise TypeError
```

Comment:
best to just `raise TypeError("Invalid constraints. Must be of MOCServerConstraints type")`; we try to avoid print statements